src/Hosting/TestHost/src/WebSocketClient.cs

```diff
@@ -76,6 +77,7 @@ public async Task<WebSocket> ConnectAsync(Uri uri, CancellationToken cancellatio
                 request.Headers.Add("Upgrade", new string[] { "websocket" });
                 request.Headers.Add("Sec-WebSocket-Version", new string[] { "13" });
                 request.Headers.Add("Sec-WebSocket-Key", new string[] { CreateRequestKey() });
+                request.Headers.Add("Sec-WebSocket-Protocol", SubProtocols.ToArray());
```

Comment:
Replace with HeaderNames constants?
https://github.com/aspnet/AspNetCore/blob/cd2983bf414a3432a96daafc50dad162432e3131/src/Http/Headers/src/HeaderNames.cs

Reply:
Yeh, can do. There's a bunch of readability improvements available to that class and the test class, I just wasn't sure how far to go so decided to go with *consistent* and see what changes get recommended.

FYI, that `HeaderNames` class has two fields for `Sec-WebSocket-Protocol`. Not sure if intentional? (`SecWebSocketProtocol` and `WebSocketSubProtocols`).

Reply:
> FYI, that `HeaderNames` class has two fields for `Sec-WebSocket-Protocol`. Not sure if intentional? (`SecWebSocketProtocol` and `WebSocketSubProtocols`).

Lol. Too bad it's a public API, let's leave the duplicate for now.